src/main/webapp/app/modeling-assessment-editor/modeling-assessment-conflict/modeling-assessment-conflict.component.ts

```diff
@@ -1,45 +1,76 @@
+import { ModelingAssessmentService } from 'app/entities/modeling-assessment';
 
 @Component({
     selector: 'jhi-modeling-assessment-conflict',
     templateUrl: './modeling-assessment-conflict.component.html',
-    styleUrls: ['./modeling-assessment-conflict.component.scss'],
+    styleUrls: ['./modeling-assessment-conflict.component.scss', '../modeling-assessment-editor.component.scss'],
 })
 export class ModelingAssessmentConflictComponent implements OnInit, AfterViewInit {
     model: UMLModel;
+    mergedFeedbacks: Feedback[];
+    currentFeedbacksCopy: Feedback[];
     modelHighlightedElementIds: Set<string>;
-    conflictIndex = 0;
+    highlightColor: string;
     user: User;
 
     currentConflict: Conflict;
     conflictingResult: ConflictingResult;
     conflictingModel: UMLModel;
     conflictingModelHighlightedElementIds: Set<string>;
+    conflicts: Conflict[];
+    conflictResolutionStates: ConflictResolutionState[];
+    conflictIndex = 0;
+    conflictsAllHandled = false;
+    modelingExercise: ModelingExercise;
+    submissionId: number;
 
-    @Input() modelingExercise: ModelingExercise;
-    @Input() conflicts: Conflict[] = [];
-
-    constructor(private route: ActivatedRoute, private modelingSubmisionService: ModelingSubmissionService, private accountService: AccountService) {}
+    constructor(
+        private jhiAlertService: JhiAlertService,
+        private route: ActivatedRoute,
+        private router: Router,
+        private modelingSubmissionService: ModelingSubmissionService,
+        private modelingAssessmentService: ModelingAssessmentService,
+        private accountService: AccountService,
+    ) {}
 
     ngOnInit() {
-        if (this.conflicts) {
-            this.updateSelectedConflict();
-            this.model = JSON.parse((this.currentConflict.result.submission as ModelingSubmission).model);
-        }
+        this.route.params.subscribe(params => {
+            this.submissionId = Number(params['submissionId']);
+            this.conflicts = this.modelingAssessmentService.getLocalConflicts(this.submissionId);
+            if (this.conflicts) {
+                this.initComponent();
+                this.model = JSON.parse((this.currentConflict.causingConflictingResult.result.submission as ModelingSubmission).model);
+                this.modelingExercise = this.currentConflict.causingConflictingResult.result.participation.exercise as ModelingExercise;
+            } else {
+                this.jhiAlertService.error('modelingAssessmentEditor.messages.noConflicts');
+            }
+        });
         this.accountService.identity().then(value => (this.user = value));
     }
 
     ngAfterViewInit() {
         this.setSameWidthOnModelingAssessments();
     }
 
+    initComponent() {
+        this.mergedFeedbacks = JSON.parse(JSON.stringify(this.conflicts[0].causingConflictingResult.result.feedbacks));
```

Comment:
There is a couple of instances in this class where JSON.stringify is followed by JSON.parse.
What does this do? Shouldn't this return the initial value before the stringify?

I speculate, that you are doing this to copy the array.
If this is the reason, you can just do something like this, if you don't want to mutate the initial array:

this.mergedFeedbacks = [...this.conflicts[0].causingConflictingResult.result.feedbacks];

Something similiar for the conflictResolutionStates:
this.conflictResolutionStates = this.confllicts.map(() => ConflictResolutionState.UNHANDLED);

Reply:
Thanks for that hint! Adapted the code according to that..